pkg/lib/cockpit-components-terminal-scribery.jsx

```diff
@@ -0,0 +1,201 @@
+            onTitleChanged: React.PropTypes.func
+        },
+
+        componentWillMount: function () {
+            var term = new Term({
+                cols: this.state.cols || 80,
+                rows: this.state.rows || 25,
+                screenKeys: true,
+                useStyle: true
+            });
+
+            term.on('data', function(data) {
+                if (this.props.channel.valid)
+                    this.props.channel.send(data);
+            }.bind(this));
+
+            if (this.props.onTitleChanged)
+                term.on('title', this.props.onTitleChanged);
+
+            this.setState({ terminal: term });
+
+            window.addEventListener("keydown", this.handleKeyDown, false);
+        },
+
+        componentDidMount: function () {
+            this.state.terminal.open(this.refs.terminal);
+            this.connectChannel();
+
+            if (!this.props.rows) {
+                window.addEventListener('resize', this.onWindowResize);
+                this.onWindowResize();
+            }
+        },
+
+        componentWillUpdate: function (nextProps, nextState) {
+            if (nextState.cols !== this.state.cols || nextState.rows !== this.state.rows) {
+                this.state.terminal.resize(nextState.cols, nextState.rows);
+                this.props.channel.control({
+                    window: {
+                        rows: nextState.rows,
+                        cols: nextState.cols
+                    }
+                });
+            }
+
+            if (nextProps.channel !== this.props.channel) {
+                this.state.terminal.reset();
+                this.disconnectChannel();
+            }
+        },
+
+        componentDidUpdate: function (prevProps) {
+            if (prevProps.channel !== this.props.channel)
+                this.connectChannel();
+        },
+
+        sendToTerm: function(value) {
+            this.state.terminal.send(value);
+        },
+
+        handleKeyDown: function(event) {
+            let keyCodesFuncs = {
+                Space: ' ',
+                ArrowUp: '\x1b[A',
+                ArrowDown: '\x1b[B',
+                ArrowRight: '\x1b[C',
+                ArrowLeft: '\x1b[D',
+            };
+
+            if (keyCodesFuncs[event.code]) {
+                this.sendToTerm(keyCodesFuncs[event.code]);
+            }
+        },
+
+        stopPlayback: function() {
+            this.sendToTerm('stop');
+        },
+
+        startPlayback: function() {
+            this.sendToTerm('start');
+        },
+
+        render: function () {
+            // ensure react never reuses this div by keying it with the terminal widget
+            return (
+                <div>
+                    <div>
+                        <button type="button" onClick={this.stopPlayback}>Stop</button>
+                        <button type="button" onClick={this.startPlayback}>Start</button>
+                    </div>
```

Comment:
I might work on adding a bit of styling for these buttons and Recording in general.